### PR TITLE
Add pro up-sell to request page sidebar

### PR DESCRIPTION
**Pull request by garethrees**
## What does this do?

* Various code cleanup
* Add pro up-sell to request page sidebar

## Why was this needed?

Direct more people to `/pro` for £££.

## Screenshots

Screenshot 2019-08-02 at 14 20 11

### CODE REVIEW

File: app/views/request/_sidebar.html.erb [hunk 1]
```diff
@@ -9,32 +9,32 @@
                locals: { info_request: @info_request } %>
   <% end %>
 
-  <div class="request__special-status">
```
Comment (gbp) [inline app/views/request/_sidebar.html.erb]: I would split this into two commits, one where the indentation is sorted out, the second where the empty div is corrected. Would give a better diffs.

File: app/assets/stylesheets/responsive/_sidebar_layout.scss [hunk 2]
```diff
@@ -12,6 +12,8 @@
   }
 }
 
+/* TODO: Not sure this is still used – we use the request heade action bar for
```
Comment (gbp) [inline app/assets/stylesheets/responsive/_sidebar_layout.scss]: Typo? s/heade/header/ ?

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -213,4 +213,10 @@ def site_wide_announcement
       session[:announcement_dismissals]
     ).first
   end
+
+  def show_pro_upsell?(user)
+    return false unless feature_enabled?(:alaveteli_pro)
```
Comment (gbp) [inline app/helpers/application_helper.rb]: As we're just dealing with booleans here are guard clauses necessary, could be written as but is it as readable?

```suggestion
    feature_enabled?(:alaveteli_pro) && (!user || user && !user.is_pro?)
```

File: app/views/request/_sidebar.html.erb [hunk 4]
```diff
@@ -1,49 +1,66 @@
...
     <%= render :partial => "general/new_request" %>
   </div>
 
+  <% if show_pro_upsell?(@user) %>
+    <div class="sidebar__section pro-upsell">
+      <% pro_link = link_to account_request_index_path do %>
+        <%= AlaveteliConfiguration.pro_site_name -%>
+      <% end %>
+
+      <% pro_site_name_link = content_tag(:strong, pro_link) %>
```
Comment (gbp) [inline app/views/request/_sidebar.html.erb]: Extract to helper? Avoids variable assignment in the views and so its testable too?

File: app/views/request/_sidebar.html.erb [hunk 5]
```diff
@@ -1,49 +1,66 @@
...
+
+      <%= _('{{pro_site_name_link}} is a powerful, fully-featured FOI ' \
```
Comment (lizconlan) [inline app/views/request/_sidebar.html.erb]: yay, this exact wording is used on the account request page so we can go ahead and lock the translations down for the release when we're ready

Comment (lizconlan) [inline app/views/request/_sidebar.html.erb]: hmm, thinking in terms of translator effort and release blockers, this is the same wording as the account request page but a different variable name

https://github.com/mysociety/alaveteli/blob/develop/app/views/alaveteli_pro/account_request/index.html.erb#L4-L6

File: app/views/request/_sidebar.html.erb (hunk 1 above)
Comment (garethrees) [reply app/views/request/_sidebar.html.erb]: Will fix in merge 👍

File: app/views/request/_sidebar.html.erb (hunk 5 above)
Comment (garethrees) [reply app/views/request/_sidebar.html.erb]: Fixed in c62120410

File: app/helpers/application_helper.rb (hunk 3 above)
Comment (garethrees) [reply app/helpers/application_helper.rb]: Yeah I couldn't decide between this, but happy to reduce it down. Fixed in c7860024c.

File: app/assets/stylesheets/responsive/_sidebar_layout.scss (hunk 2 above)
Comment (garethrees) [reply app/assets/stylesheets/responsive/_sidebar_layout.scss]: Fixed in 99507c4e8

File: app/views/request/_sidebar.html.erb (hunk 4 above)
Comment (garethrees) [reply app/views/request/_sidebar.html.erb]: I couldn't decide about this either. I was thinking the whole thing should be a helper; something like:

```ruby
<= pro_upsell if show_pro_upsell?(@user) %>
```

But, if there's success here, I'd like to also add it to one of the slots under the search bar in `/select_authority`, which would be a good time to refactor common bits? I didn't really want to delay this while I think about that.

I can't really think of a good name for a helper that does just the link at the moment. Any thoughts? It seems like it needs to be something like `strong_pro_site_name_link_to_account_request_index` to be explicit about what its doing, but that's rather unwieldy!

Comment (garethrees) [reply app/views/request/_sidebar.html.erb]: I guess there could be a helper for just the text part:

```erb
<% if show_pro_upsell?(@user) %>
  <div class="sidebar__section pro-upsell">
    <%= pro_upsell %>
  </div>
<% end %>
```

Comment (garethrees) [reply app/views/request/_sidebar.html.erb]: Fixed in a91111a7d.

**Pull request merged**